File: shelf/src/test/java/com/toddway/shelf/BaseTest.java
package com.toddway.shelf;

import org.junit.Before;

import java.io.File;
import java.util.List;

import io.reactivex.observers.TestObserver;
import io.reactivex.subscribers.TestSubscriber;

/**
 * Created by tway on 1/11/17.
 */

public class BaseTest {

    ShelfItem item;
    TestObserver<String> subscriber;
    Shelf shelf;
    String newValue;
    String cacheValue;

    @Before
    public void beforeEach() {
        shelf = new Shelf(new File("/tmp"));
        item = shelf.item("string");
        subscriber = new TestObserver<>();
        newValue = "new value";
        cacheValue = "cache value";
    }

    public void givenNoCache() {
        item.clear();
    }

    public void givenValidCache() {
        item.maxAge(5000).put(cacheValue);
    }

    public void givenInvalidCache() {
        item.maxAge(0).put(cacheValue);
    }

    public void givenNoNew() {
        newValue = null;
    }

    public <T> void printValues(List<T> values) {
        System.out.println("values...");
        for (T value : values) System.out.println("value: " + value);
    }

}

File: shelf/src/test/java/com/toddway/shelf/BaseTest.kt
package com.toddway.shelf

import io.reactivex.observers.TestObserver
import org.junit.Before
import java.io.File

/**
 * Created by tway on 1/11/17.
 * updated nschwermann 12/21/17
 */

abstract class BaseTest {

    lateinit var item: ShelfItem
    lateinit var subscriber: TestObserver<String>
    lateinit var shelf: Shelf
    var newValue: String? = null
    lateinit var cacheValue: String

    @Before
    open fun beforeEach() {
        shelf = Shelf(File("/tmp"))
        item = shelf.item("string")
        subscriber = TestObserver()
        newValue = "new value"
        cacheValue = "cache value"
    }

    fun givenNoCache() {
        item.clear()
    }

    fun givenValidCache() {
        item.maxAge(5000).put(cacheValue)
    }

    fun givenInvalidCache() {
        item.maxAge(0).put(cacheValue)
    }

    fun givenNoNew() {
        newValue = null
    }

    fun <T> printValues(values: List<T>) {
        for (value in values) println("value: " + value)
    }

}
